tsgen/__main__.py

```diff
@@ -66,11 +71,11 @@ def generate_c_code(self):
         Return:
             Tuple containing paths to C header and source files
         """
-        HEADER_FILE = f'{self.C_DATABASE_NAME}.h'
-        SOURCE_FILE = f'{self.C_DATABASE_NAME}.c'
+        HEADER_FILE = f'{self.C_DATABASE_NAME + bus}.h'
+        SOURCE_FILE = f'{self.C_DATABASE_NAME + bus}.c'
 
-        header, source, _, _ = do_generate_c_code(self._dbc,
-                                                  self.C_DATABASE_NAME, 
+        header, source, _, _ = do_generate_c_code(self.__getattribute__(bus),
```

Comment:
What does `self.__getattribute(bus)` do? Not suggesting it's wrong, just never seen it before.

Reply:
I think it's a way to convert `bus` string into variable name (equal to value of `bus`). In this example we want to use variable (self.can_s) or (self.can_c) as argument to function `do_generate_c_code`, `__getattribute__` looks for the variable named `can_s` or `can_c`.

Reply:
We still need `bus` for specifying HEADER_FILE before in this function, so I think its a nice method to do this.

Reply:
I guess it's neat, but I think it's quite hard to understand and future people working on this code will not get it. I personally think the DBC variable should just be passed as an argument to this function like:

```python
def generate_c_code(self, bus_name, dbc):
```

There's also the problem that if the DBC variable were given a different name to the output source code file, it would break the code.

Reply:
In fact this gives me a better idea, to make the system more flexible:
- Use the DBC file name to create the source code file name, instead of defining it as a constant. Like now we have `CAN_S_APP = 'can_s'` but we could just generate this from the DBC file name (and just replace any non alphanumeric characters with underscores).
- This also has the advantage of expanding to as many (or few) DBCs as we want. So we could choose to make a new release which only updates from one of the databases, whereas now you have to provide both. So when the script starts, it just sees how many DBC files it was given as an argument and iterates over them to generate the unpack functions and OCT handlers.

Does that make sense? I think it would be a nice improvement if you have time, if not it can wait and become an issue.